source/core/Mouse.cpp

```diff
@@ -16,15 +16,22 @@
 
 Window *Mouse::s_window = nullptr;
 
-void Mouse::resetToWindowCenter() {
-	SDL_WarpMouseInWindow(s_window->window(), SCREEN_WIDTH / 2, SCREEN_HEIGHT / 2);
+Vector2i Mouse::resetToWindowCenter() {
+	const Vector2i newPos(SCREEN_WIDTH / 2, SCREEN_HEIGHT / 2);
+	sf::Mouse::setPosition(newPos, *s_window->window());
+//	sf::Mouse::setPosition({s_window->window()->getSize() / 2}, *s_window->window());
+	return newPos;
+}
```

Comment:
I think the return value is useless here, but you maybe had a reason to add it?

And using directly `window().getSize()` can be indeed better.

Reply:
I don't fully remember, I'll look into it later maybe.
I think I had to add the return value, because otherwise I would have had to check for the mouse position unnecessarily in `GameState::onEvent()` before resetting, which might be slower. In SDL that was easier I think.